modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1678,7 +1901,10 @@ public IgniteFutureImpl<Void> restoreSnapshot(String name, @Nullable String snpP
         for (File tmp : snapshotTmpDir().listFiles())
             U.delete(tmp);
 
-        deleteSnapshot(snpDir, pdsSettings.folderName());
+        if (INC_SNP_NAME_PATTERN.matcher(snpDir.getName()).matches() && snpDir.getAbsolutePath().contains(INC_SNP_DIR))
```

Comment:
As I can see, it's enough to check that `snpDir` contains `increments` path segment, no need check name pattern actually.

Reply:
What if full snapshot has name "increments"?

Reply:
Path should contain segment "increments", like (snp/increments/000...).

Reply:
Don't think anything should be fixed here. Looke like current check works)